engine/stagemaster/src/lib.rs

```diff
@@ -24,25 +26,32 @@ pub mod camera_control;
 
 use kay::{ID, ActorSystem, Fate, External};
 use descartes::{N, P2, V2, P3, Into2d, Shape};
-use monet::{RendererID, RenderableID, SceneDescription, GlutinFacade};
-use monet::glium::glutin::{Event, MouseScrollDelta, ElementState, MouseButton};
+use monet::{RendererID, ProjectionRequesterID, RenderableID, SceneDescription, Display};
+use monet::glium::glutin::{ContextBuilder, Event, WindowBuilder, WindowEvent, MouseScrollDelta,
+                           ElementState, MouseButton, KeyboardInput};
+use monet::glium::glutin::EventsLoop;
 pub use monet::glium::glutin::VirtualKeyCode;
 use geometry::AnyShape;
 use std::collections::{HashMap, HashSet};
-use imgui::{ImGui, ImVec4, ImGuiSetCond_FirstUseEver, ImGuiKey};
-use imgui_sys::{ImFontConfig, ImGuiCol, ImGuiAlign_Center, ImFontConfig_DefaultConstructor};
-use imgui::glium_renderer::Renderer as ImguiRenderer;
+use imgui::{ImGui, ImVec2, ImVec4, ImGuiSetCond_FirstUseEver, ImGuiKey};
+use imgui_sys::{ImFontConfig, ImGuiCol, ImFontConfig_DefaultConstructor};
+use imgui_glium_renderer::Renderer as ImguiRenderer;
 use std::collections::BTreeMap;
 
 pub struct UserInterface {
-    window: GlutinFacade,
+    events_loop: EventsLoop,
+    interactables: HashMap<ID, (AnyShape, usize)>,
+    ui_inner: UserInterfaceInner,
+}
+
+struct UserInterfaceInner {
```

Comment:
Why did you split it into `UserInterface` and `UserInterfaceInner`? And why are `interactables` not in `UserInterfaceInner`?

Reply:
fixed, thanks!

Reply:
it occured to me that it was just possible to get all events from the event loop and there is no need to have a 'large' lambda, just as it was previously.

Reply:
so there is no need for in the first place splitting!